### PR TITLE
spec: write module spec

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

This is for me to get an understanding of what is in the repo and for others to understand what the module does internally.

### CODE REVIEW

File: module/x/peggy/spec/02_state.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+### OutgoingTxBatch
+
+Stored in two possible ways, first with a height and second without (unsafe)
+
+### ValidatorSet
+
+This is the validator set of the bridge.
+
+Stored in two possible ways, first with a height and second without (unsafe)
```
Comment (jackzampolin) [inline module/x/peggy/spec/02_state.md:25]: The unsafe method is primarily in there for testing and state exports/imports.

File: module/x/peggy/spec/02_state.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+and defines overall functioning of the staking module.
+
+- Params: `Paramsspace("peggy") -> legacy_amino(params)`
+
++++ <https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/genesis.proto#L72-L104>
+
+
+### OutgoingTxBatch
+
+Stored in two possible ways, first with a height and second without (unsafe)
```
Comment (jackzampolin) [inline module/x/peggy/spec/02_state.md:19]: The unsafe method is primarily in there for testing and state exports/imports.

**Review by jackzampolin — COMMENTED** [review]
Looks good so far @marbar3778! Nice to have a spec.

File: module/x/peggy/spec/01_definitions.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+This section outlines terminology used throughout the spec and code.
+
+### Operator
+
+This is a person (or people) who control a Cosmos SDK validator node. This is also called `valoper` or "Validator Operator" in the Cosmos SDK staking section
+
+### Relayer
+
+This is a type of node that submits updates to the Peggy contract on Ethereum. It earns fees from the transactions in a batch.
```
Comment (fedekunze) [inline module/x/peggy/spec/01_definitions.md:16]: and viceversa

**Comment by marbar3778** [comment]
We can merge this without the state transition section. Fede and I will collaborate on reworking the spec to adhere to the new spec and define a plan of attack.

**Review by fedekunze — APPROVED** [review]
a few sections are unfinished but this is a great start

File: module/x/peggy/keeper/msg_server.go [hunk 4]
```diff
@@ -153,6 +153,7 @@ func (k msgServer) RequestBatch(c context.Context, msg *types.MsgRequestBatch) (
 		return nil, err
 	}
 
+	// question: is this right? If i can delegate my voting power to a different key then this would fail each time i call it
```
Comment (jkilpatr) [inline module/x/peggy/keeper/msg_server.go:156]: it's being used on the testnet, works with the delegate key, have not tried it with the validator key directly. So I think this is just confusingly written.

**Review by jkilpatr — APPROVED** [review]
nice work, going to go ahead and land this.

File: module/x/peggy/keeper/msg_server.go (hunk 4 above)
Comment (marbar3778) [reply module/x/peggy/keeper/msg_server.go:156]: ah yes, I left some questions in there, I can remove them

File: module/x/peggy/types/events.go [hunk 5]
```diff
@@ -3,15 +3,12 @@ package types
 const (
 	EventTypeObservation               = "observation"
 	EventTypeOutgoingBatch             = "outgoing_batch"
-	EventTypeMultisigBootstrap         = "multisig_bootstrap"
 	EventTypeMultisigUpdateRequest     = "multisig_update_request"
 	EventTypeOutgoingBatchCanceled     = "outgoing_batch_canceled"
 	EventTypeOutgoingLogicCallCanceled = "outgoing_logic_call_canceled"
 	EventTypeBridgeWithdrawalReceived  = "withdrawal_received"
-	EventTypeBridgeDepositReceived     = "deposit_received"
```
Comment (jkilpatr) [inline module/x/peggy/types/events.go:11]: this may not be used but deposit_received is a valid event, I'll correct this during my rebase and merge

Comment (jkilpatr) [reply module/x/peggy/types/events.go:11]: if you could open another pull to fix the events that would be a good place to correct this and is much needed.

Comment (marbar3778) [reply module/x/peggy/types/events.go:11]: Should we be keeping dead code around?

Generally I'd say no, but unsure how you approach it?

**Comment by jkilpatr** [comment]
This has been merged. git doesn't detect it as merged due to my local rebase and conflict resolution.

**Pull request closed without merging**